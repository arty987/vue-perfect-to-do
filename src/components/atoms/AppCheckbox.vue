<script>
export default {
  name: "AppCheckbox",
  props: {
    disabled: {
      type: Boolean,
      default: false
    },
    checked: {
      type: Boolean,
      default: false
    }
  }
}
</script>

<template>
  <input
      type="checkbox"
      :disabled="disabled"
      :checked="checked"
  />
</template>