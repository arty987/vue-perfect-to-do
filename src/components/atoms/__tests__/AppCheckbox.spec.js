import {shallowMount} from '@vue/test-utils'
import AppCheckbox from '@/components/atoms/AppCheckbox.vue';

describe('AppCheckbox.vue', () => {
    it('Should render AppCheckbox with default props', () => {
        const wrapper = shallowMount(AppCheckbox);
        expect(wrapper.element).toMatchSnapshot();
    });

    it('Should render AppCheckbox with props set', () => {
        const wrapper = shallowMount(AppCheckbox, {
            props: {
                disabled: true,
                checked: true
            }
        });
        expect(wrapper.element).toMatchSnapshot();
    });
})
