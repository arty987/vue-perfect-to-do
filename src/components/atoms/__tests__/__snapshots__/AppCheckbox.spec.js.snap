// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppCheckbox.vue Should render AppCheckbox with default props 1`] = `
<input
  class="app-checkbox"
  type="checkbox"
/>
`;

exports[`AppCheckbox.vue Should render AppCheckbox with props set 1`] = `
<input
  checked=""
  class="app-checkbox"
  disabled=""
  type="checkbox"
/>
`;
